pkg/server/plugin/upstreamauthority/gcpcas/gcpcas.go

```diff
@@ -0,0 +1,341 @@
+		}
+
+		allCerts = append(allCerts, page...)
+		if nextPageToken == "" {
+			break
+		}
+	}
+
+	return allCerts, nil
+}
+
+func (p *Plugin) mintX509CA(ctx context.Context, csr []byte, preferredTtl int32) (*upstreamauthority.MintX509CAResponse, error) {
+	csrParsed, err := x509.ParseCertificateRequest(csr)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse CSR: %v", err)
+	}
+
+	validity := time.Second * time.Duration(preferredTtl)
+
+	pcaClient, err := getClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	allCertRoots, err := loadCertificateAuthorities(ctx, pcaClient, p.c.RootSpec)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to load root CAs: %v.", err)
+	}
+	if len(allCertRoots) == 0 {
+		return nil, fmt.Errorf("No certificate authorities found with label pair %s:%s.", p.c.RootSpec.LabelKey, p.c.RootSpec.LabelValue)
+	}
+
+	// The certs get returned in issuance order, and we want the oldest, so just grab the first one and use that.
+	chosenCA := allCertRoots[0]
+
+	// The rest of them are still trusted
+	trustBundle := make([]*privatecapb.CertificateAuthority, 0)
+	if len(allCertRoots) > 1 {
+		trustBundle = append(trustBundle, allCertRoots[1:]...)
+	}
+
+	for _, spec := range p.c.TrustBundleSpecs {
+		trustBundleCerts, err := loadCertificateAuthorities(ctx, pcaClient, spec)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to load trust bundle CAs: %v.", err)
+		}
+		trustBundle = append(trustBundle, trustBundleCerts...)
+	}
+
+	parentPath := chosenCA.Name
+	p.log.Info("Minting X509 intermediate CA ", "ca-certificate", parentPath, "ttl ", validity)
+
+	subject := privatecapb.Subject{}
+	extractFirst := func(strings []string, into *string) {
+		if len(strings) > 0 {
+			*into = strings[0]
+		}
+	}
+
+	extractFirst(csrParsed.Subject.Organization, &subject.Organization)
+	extractFirst(csrParsed.Subject.OrganizationalUnit, &subject.OrganizationalUnit)
+	extractFirst(csrParsed.Subject.Locality, &subject.Locality)
+	extractFirst(csrParsed.Subject.Province, &subject.Province)
+
+	createRequest := privatecapb.CreateCertificateRequest{
+		Parent: parentPath,
+		Certificate: &privatecapb.Certificate{
+			Lifetime: ptypes.DurationProto(validity),
+			CertificateConfig: &privatecapb.Certificate_Config{
+				Config: &privatecapb.CertificateConfig{
+					PublicKey: &privatecapb.PublicKey{
+						Type: privatecapb.PublicKey_PEM_EC_KEY,
+						Key: pem.EncodeToMemory(
+							&pem.Block{
+								Type:  "PUBLIC KEY",
+								Bytes: csrParsed.RawSubjectPublicKeyInfo,
+							},
+						),
+					},
+					SubjectConfig: &privatecapb.CertificateConfig_SubjectConfig{
+						Subject:    &subject,
+						CommonName: csrParsed.Subject.CommonName,
+					},
+					ReusableConfig: &privatecapb.ReusableConfigWrapper{
+						ConfigValues: &privatecapb.ReusableConfigWrapper_ReusableConfig{
+							ReusableConfig: fmt.Sprintf("projects/privateca-data/locations/%s/reusableConfigs/subordinate-mtls-pathlen-0", p.c.RootSpec.Location),
```

Comment:
is this magic string documented anywhere? maybe we can make it a const and link to some docs on it?